examples/cds-feature-attachments-oss/src/main/java/com/sap/cds/feature/attachments/oss/client/OSServiceImpl.java

```diff
@@ -0,0 +1,67 @@
+package com.sap.cds.feature.attachments.oss.client;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.time.Instant;
+
+import com.sap.cds.feature.attachments.service.model.service.AttachmentModificationResult;
+import com.sap.cds.feature.attachments.service.model.service.MarkAsDeletedInput;
+import com.sap.cds.services.ServiceDelegator;
+import com.sap.cloud.environment.servicebinding.api.ServiceBinding;
+
+/**
+ * Implementation of the {@link OSService} interface.
+ * This class handles the readAttchment, createAttachment, markAttachmentAsDeleted,
+ * and restoreAttachment methods, delegating the actual logic to the appropriate
+ * service methods.
+ */
+public class OSServiceImpl extends ServiceDelegator implements OSService {
```

Comment:
It's not required to implement a service, instead you should implement handler methods for the emitted events of the AttachmentService. For example, have a look at the FS based handlers: https://github.com/cap-java/cds-feature-attachments/blob/main/examples/cds-feature-attachments-fs/src/main/java/com/sap/cds/feature/attachments/fs/handler/FSAttachmentsServiceHandler.java